commit 8712836b30cef5d49bc3bb8bc3da88a40e11e574
Author: David Hildenbrand <dahi@linux.vnet.ibm.com>
Date:   Mon Mar 3 10:55:13 2014 +0100

    KVM: s390: deliver program irq parameters and use correct ilc
    
    When a program interrupt was to be delivered until now, no program interrupt
    parameters were stored in the low-core of the target vcpu.
    
    This patch enables the delivery of those program interrupt parameters, takes
    care of concurrent PER events which can be injected in addition to any program
    interrupt and uses the correct instruction length code (depending on the
    interception code) for the injection of program interrupts.
    
    Signed-off-by: David Hildenbrand <dahi@linux.vnet.ibm.com>
    Signed-off-by: Christian Borntraeger <borntraeger@de.ibm.com>

diff --git a/arch/s390/include/asm/kvm_host.h b/arch/s390/include/asm/kvm_host.h
index b8c808192893..5e5a14db8c21 100644
--- a/arch/s390/include/asm/kvm_host.h
+++ b/arch/s390/include/asm/kvm_host.h
@@ -102,6 +102,12 @@ struct kvm_s390_sie_block {
 #define ICTL_RRBE 0x00001000
 	__u32	ictl;			/* 0x0048 */
 	__u32	eca;			/* 0x004c */
+#define ICPT_INST	0x04
+#define ICPT_PROGI	0x08
+#define ICPT_INSTPROGI	0x0C
+#define ICPT_OPEREXC	0x2C
+#define ICPT_PARTEXEC	0x38
+#define ICPT_IOINST	0x40
 	__u8	icptcode;		/* 0x0050 */
 	__u8	reserved51;		/* 0x0051 */
 	__u16	ihcpu;			/* 0x0052 */